ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/Invoker.java

```diff
@@ -163,6 +241,168 @@ protected PackageCompilation compile(Project project) throws InvokerException {
+                addDiagnostic(Diagnostic.error("Import resolution failed. Module not found."));
+                throw new InvokerException();
+            }
+        }
+    }
+
+    /* Execution methods */
+
+    /**
+     * Executes a compiled project.
+     * It is expected that the project had no compiler errors.
+     * The process is run and the stdout is collected and printed.
+     *
+     * @param jBallerinaBackend Backed to use.
+     * @return Whether process execution was successful.
+     * @throws InvokerException If execution failed.
+     */
+    protected boolean executeProject(JBallerinaBackend jBallerinaBackend) throws InvokerException {
+        if (bufferFile == null) {
+            throw new UnsupportedOperationException("Buffer file must be set before execution");
+        }
+
+        // Main method class name is file name without extension
+        String fileName = bufferFile.getName();
+        String mainMethodClassName = fileName.substring(0, fileName.length() - TEMP_FILE_SUFFIX.length());
+
+        JarResolver jarResolver = jBallerinaBackend.jarResolver();
+        ClassLoader classLoader = jarResolver.getClassLoaderWithRequiredJarFilesForExecution();
+        // First run configure initialization
+        invokeDirectMethod(classLoader, CONFIGURE_INIT_CLASS_NAME, CONFIGURE_INIT_METHOD_NAME,
+                new Class[]{Path.class}, new Object[]{CONFIG_PATH});
+        // Initialize the module
+        invokeScheduledMethod(classLoader, MODULE_INIT_CLASS_NAME, MODULE_INIT_METHOD_NAME);
+        // Then call main method
+        Object result = invokeScheduledMethod(classLoader, mainMethodClassName, MODULE_MAIN_METHOD_NAME);
+        addDiagnostic(Diagnostic.debug("Result: " + result));
+        return result == null;
+    }
+
+    /* Invocation methods */
+
+    /**
+     * Invokes a method that is in the given class.
+     * The method must be a static method accepting only one parameter, a {@link Strand}.
+     *
+     * @param classLoader Class loader to find the class.
+     * @param className   Class name with the method.
+     * @param methodName  Method name to invoke.
+     * @return The result of the invocation.
+     * @throws InvokerException If invocation failed.
+     */
+    protected Object invokeScheduledMethod(ClassLoader classLoader, String className, String methodName)
+            throws InvokerException {
+        try {
+            addDiagnostic(Diagnostic.debug(String.format("Running %s.%s on schedule", className, methodName)));
+
+            // Get class and method references
+            Class<?> clazz = classLoader.loadClass(className);
+            Method method = clazz.getDeclaredMethod(methodName, Strand.class);
+            Function<Object[], Object> methodInvocation = createInvokerCallback(method);
+
+            // Schedule and run the function and return if result is valid
+            BFuture out = scheduler.schedule(new Object[1], methodInvocation, null, null, null,
```

Comment:
Use the apis in `io.ballerina.runtime.api.Runtime` to invoke methods. The methods in the schedulers are not supposed to be used

Reply:
These are not object methods per-se. They are `main`, `$moduleInit` and `$configureInit` functions.

Reply:
since there's no access to strand in your case it is okay to create a `new Scheduler()`